### ISSUE TITLE
DataObject Class

### ISSUE THREAD

**Issue by mattetti** [microsoft/FluidFramework#3512]
When looking at the TSDoc doc the DataObject class, I noticed a weird section about Generics:

```
Generics: P - represents a type that will define optional providers that will be injected S - the initial state type that the produced data store may take during creation E - represents events that will be available in the EventForwarder
```

https://fluidframework.com/apis/aqueduct/dataobject/

**Comment by vladsud** [comment]
Links changed, here is the one: https://fluidframework.com/docs/apis/aqueduct/dataobject/
I see only this for O, and nothing for other arguments:

Generics: O - represents a type that will define optional providers that will be injected S - the initial state type that the produced data store may take during creation E - represents events that will be available in the EventForwarder

Here is how it looks in the code:
```typescript
/**
 * DataObject is a base data store that is primed with a root directory. It
 * ensures that it is created and ready before you can access it.
 *
 * Having a single root directory allows for easier development. Instead of creating
 * and registering channels with the runtime any new DDS that is set on the root
 * will automatically be registered.
 *
 * Generics:
 * O - represents a type that will define optional providers that will be injected
 * S - the initial state type that the produced data store may take during creation
 * E - represents events that will be available in the EventForwarder
 */
```

@tylerbutler , it sounds like there is some bug that drops content, or we should write comments differently.
Any chance you can take a look?

**Comment by tylerbutler** [comment]
This will be addressed by PR #6546.
